crates/transaction-pool/src/blobstore/disk.rs

```diff
@@ -279,15 +280,17 @@ impl DiskFileBlobStoreInner {
     }
 
     /// Retrieves the blob for the given transaction hash from the blob cache or disk.
-    fn get_one(&self, tx: B256) -> Result<Option<BlobTransactionSidecar>, BlobStoreError> {
+    fn get_one(&self, tx: B256) -> Result<Option<Arc<BlobTransactionSidecar>>, BlobStoreError> {
         if let Some(blob) = self.blob_cache.lock().get(&tx) {
             return Ok(Some(blob.clone()))
         }
         let blob = self.read_one(tx)?;
+
         if let Some(blob) = &blob {
-            self.blob_cache.lock().insert(tx, blob.clone());
+            self.blob_cache.lock().insert(tx, Arc::new(blob.clone()));
```

Comment:
this should be Arc'ed here, otherwise we clone this twice

Reply:
I am not sure what this means 😕